packages/fake-driver/lib/scripts/fake-stdin.js

```diff
@@ -0,0 +1,16 @@
+import readline from 'node:readline';
+
+const rl = readline.createInterface({input: process.stdin, output: process.stderr});
+
+rl.question('Press ENTER to continue: ', () => {
+  rl.close();
+  // eslint-disable-next-line no-console
+  console.error('You did it!');
+});
+
+rl.once('close', () => {
+  // there is some sort of weirdness where Node.js doesn't know when to stop listening on `process.stdin` if
```

Comment:
perhaps we just need to feed `eof` into the stream?
Are you sure `unref`'ing it won't create a memory leak?

Reply:
```
const { spawn } = require('node:child_process');
const ps = spawn('ps', ['ax']);
const grep = spawn('grep', ['ssh']);

ps.stdout.on('data', (data) => {
  grep.stdin.write(data);
});

ps.stderr.on('data', (data) => {
  console.error(`ps stderr: ${data}`);
});

ps.on('close', (code) => {
  if (code !== 0) {
    console.log(`ps process exited with code ${code}`);
  }
  grep.stdin.end();
});

grep.stdout.on('data', (data) => {
  console.log(data.toString());
});

grep.stderr.on('data', (data) => {
  console.error(`grep stderr: ${data}`);
});

grep.on('close', (code) => {
  if (code !== 0) {
    console.log(`grep process exited with code ${code}`);
  }
});
```

An example from https://nodejs.org/api/child_process.html#child_processspawncommand-args-options

Reply:
Why would `unref` cause a memory leak?  There's a bug around this, which is why I did what I did: https://github.com/nodejs/node/issues/32291

Reply:
Anyway, the `end()` seems to work in lieu of `unref()`, so I'll just use that.